cgo/proofs.go

```diff
@@ -169,13 +169,13 @@ func GenerateWinningPoSt(randomness *ByteArray32, replicas SliceRefPrivateReplic
 func GenerateWindowPoSt(randomness *ByteArray32, replicas SliceRefPrivateReplicaInfo, proverId *ByteArray32) ([]PoStProofGo, []uint64, error) {
 	resp := C.generate_window_post(randomness, replicas, proverId)
 	defer resp.destroy()
-	faults := resp.value.faulty_sectors.copy()
-
 	if err := CheckErr(resp); err != nil {
+		faults := resp.value.faulty_sectors.copy()
```

Comment:
This is invalid memory otherwise? Maybe we should fix it in rust?

Reply:
it wasn’t an issue, just wanted to make the code more defensive on both sides